### PR TITLE
Fix various bugs in serialization and computation of local AABB

### PR DESCRIPTION
**Pull request by jcarpent**

### CODE REVIEW

File: cmake-modules/Findassimp.cmake [hunk 1]
```diff
@@ -1,7 +1,26 @@
...
+      set(_ASSIMP_TARGET_OK FALSE)
+    endif()
+  endforeach()
+  if(NOT _ASSIMP_TARGET_OK)
+    get_target_property(_ASSIMP_IMP_LOC_RELEASE assimp::assimp IMPORTED_LOCATION_RELEASE)
+    set_target_properties(assimp::assimp PROPERTIES
+      INTERFACE_INCLUDE_DIRECTORIES "${ASSIMP_INCLUDE_DIRS}"
+      IMPORTED_LOCATION "${_ASSIMP_IMP_LOC_RELEASE}")
+  endif()
+else()
```
Comment (jcarpent) [inline cmake-modules/Findassimp.cmake:23]: Should we move this part to the JRL cmake module for unification?

File: include/hpp/fcl/serialization/octree.h [hunk 2]
```diff
@@ -77,17 +69,19 @@ void load(Archive &ar, hpp::fcl::OcTree &octree,
...
   Accessor &access = reinterpret_cast<Accessor &>(octree);
 
+  std::size_t tree_data_size;
+  ar >> make_nvp("tree_data_size", tree_data_size);
+
   std::string stream_str;
-  ar >> make_nvp("tree_data", stream_str);
+  stream_str.resize(tree_data_size);
+  /// TODO use stream_str.data in C++17
+  ar >> make_nvp("tree_data", make_array(&stream_str[0], tree_data_size));
```
Comment (jcarpent) [inline include/hpp/fcl/serialization/octree.h:78]:
```suggestion
  if(tree_data_size > 0)
    ar >> make_nvp("tree_data", make_array(&stream_str[0], tree_data_size));
```

File: include/hpp/fcl/serialization/octree.h [hunk 3]
```diff
@@ -35,10 +42,12 @@ void save(Archive &ar, const hpp::fcl::OcTree &octree,
...
-  ar << make_nvp("tree_data", stream_str);
-
-  // const double resolution = octree.getResolution();
-  // ar << make_nvp("resolution", resolution);
+  auto size = stream_str.size();
+  // We can't directly serialize stream_str because it contains binary data.
+  // This create a bug on Windows with text_archive
+  ar << make_nvp("tree_data_size", size);
+  ar << make_nvp("tree_data",
+                 make_array(stream_str.c_str(), stream_str.size()));
```
Comment (jcarpent) [inline include/hpp/fcl/serialization/octree.h:50]:
```suggestion
  if(size > 0)
    ar << make_nvp("tree_data",
                   make_array(stream_str.c_str(), size));
```

File: cmake-modules/Findassimp.cmake (hunk 1 above)
Comment (jorisv) [reply cmake-modules/Findassimp.cmake:23]: I will move it

File: include/hpp/fcl/serialization/octree.h (hunk 2 above)
Comment (jcarpent) [inline include/hpp/fcl/serialization/octree.h:78]:
```suggestion
  assert(tree_data_size > 0 && "tree_data_size should be greater than 0");
  ar >> make_nvp("tree_data", make_array(&stream_str[0], tree_data_size));
```

**Pull request merged**